### ISSUE TITLE
Cách xử lý ảnh đầu vào để train theo batch như thế nào ?

### ISSUE THREAD

**Issue by buiquangmanhhp1999** [pbcquoc/vietocr#8]
Em chào anh.

Trước hết em cảm ơn anh vì thư viện VietOCR thực sự có ích cho cộng đồng học máy nói chung và em nói riêng. Trong quá trình tìm hiểu thư viện VietOcr, em có một thắc mắc nhỏ là em chưa tìm thấy cách xử lý ảnh line text đầu vào có chiều dài khác nhau để có thể train được theo từng batch. Nếu có thể, anh có thể chia sẻ giúp em được không ?

Em cảm ơn anh. Chúc anh trong thời gian tới, công việc tiếp tục phát triển và có nhiều đóng góp cho cộng đồng AI Việt Nam hơn nữa.

**Comment by pbcquoc** [comment]
Hi e, thư viện này là train theo từng batch e nhé. 
Để train theo từng batch thì đơn giản là e chỉ cần lấy những ảnh có kích thước bằng nhau gộp thành một batch rồi forward qua thôi. 
Đơn giản là e sẽ tính chiều dài của mỗi ảnh, sắp sếp trước các ảnh có cùng width thì cho vào cùng một bucket rồi sao đó sampling từ các bucket.

Em có thể tham khảo file này vietocr/loader/DataLoader.py

**Comment by buiquangmanhhp1999** [comment]
Nếu như trong data có nhiều line text có chiều rộng khác nhau quá, và số
lượng ảnh có cùng một chiều rộng ít hơn batch size. Anh có thêm padding
hoặc resize hay có phương pháp xử lý khác không ạ

Vào Th 7, 26 thg 9, 2020 lúc 08:05 Pham Ba Cuong Quoc <
notifications@github.com> đã viết:

**Comment by pbcquoc** [comment]
e có thể resize width về bucket gần nhất mà e quy định nhé. 
ví dụ e có 6 bucket 50, 60, 70, 80, 90, 100 thì với một ảnh cho trước e sẽ resize width về bucket về gần để giảm biến dạng ảnh. 
anh không xử dụng padding

**Comment by buiquangmanhhp1999** [comment]
Em cảm ơn anh ạ😃

Vào Th 7, 26 thg 9, 2020 lúc 12:17 Pham Ba Cuong Quoc <
notifications@github.com> đã viết:
